src/main/java/net/dries007/tfc/api/recipes/BarrelRecipe.java

```diff
@@ -122,4 +124,14 @@ private int getMultiplier(FluidStack inputFluid, ItemStack inputStack)
         }
         return 0;
     }
+
+    public boolean shouldRepeate()
+    {
+        return false;
+    }
+
+    public void playSound(World world, BlockPos pos, int tickCounter)
```

Comment:
This seems like really specific design that will be useful in only one case. I don't like the current implementation - if there's a way to make this actually expandable and useful, i.e. a generic `onRecipeComplete` function then I'll be okay with it, but if it's just an exception for one recipe, I'd rather remove it.

Reply:
Yeah, I agree.